rotkehlchen/chain/evm/decoding/compound/v3/decoder.py

```diff
@@ -106,47 +110,75 @@ def decode_reward_claim(self, context: DecoderContext) -> DecodingOutput:
 
         return DEFAULT_DECODING_OUTPUT
 
-    def _decode_supply_or_withdraw_event(
+    def _correct_supply_or_withdraw_event(
+            self,
+            decoded_events: list['EvmEvent'],
+            transaction: 'EvmTransaction',  # pylint: disable=unused-argument
+            all_logs: list['EvmTxReceiptLog'],  # pylint: disable=unused-argument
+    ) -> list['EvmEvent']:
+        """Post decoding function to correct supply/withdraw event's receive amount in notes."""
+        for event in decoded_events:
+            if ((
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.RECEIVE_WRAPPED
+            ) or (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.RETURN_WRAPPED
+            )) and event.counterparty == CPT_COMPOUND_V3 and event.notes is not None:
+                event.notes = event.notes.format(amount=event.balance.amount)  # set the amount
+                break
+        return decoded_events
+
+    def _decode_supply_or_repay_event(
             self,
             context: DecoderContext,
-            compound_token: EvmToken,
+            compound_token: 'EvmToken',
     ) -> DecodingOutput:
-        """Decode a compound v3 supply or withdraw event. Takes decoder context and
-        the compound v3 wrapped token of the supplied/withdrawn underlying token"""
+        """Decode a compound v3 supply or repay event. Takes decoder context and
+        the compound v3 wrapped token of the supplied/withdrawn underlying token."""
         if (underlying_token := self._get_compound_underlying_token(compound_token)) is None:
             log.error(
                 f'At compound v3 supply/withdraw decoding of tx '
                 f'{context.transaction.tx_hash.hex()} the underlying token was not found.',
             )
             return DEFAULT_DECODING_OUTPUT
 
+        receiving_ctoken = False
+        for tx_log in context.all_logs:
+            if (
+                tx_log.address == compound_token.evm_address and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                hex_or_bytes_to_address(tx_log.topics[1]) == ZERO_ADDRESS and  # from
+                hex_or_bytes_to_address(tx_log.topics[2]) == hex_or_bytes_to_address(context.tx_log.topics[2])  # to  # noqa: E501
+            ):
+                receiving_ctoken = True
```

Comment:
Hello @OjusWiZard , what about the compound-v3 transactions that are not generating any ctoken? I have seen in some compound transaction of [WETH contract](https://etherscan.io/address/0xA17581A9E3356d9A858b789D68B4d866e593aE94) that there is no ctoken, generated or transferred, for the supply/borrow/repay/withdraw operations. I am not sure if I fully got this code section, but it seems to me that this line can be reached even if there is any other protocol token generation in the same transaction (and maybe this is the reason it is working with the transactions I tried, even if there is no ctoken generation there)

Reply:
yes, It will only be true when new compound tokens are minted and transferred to the tracked user. That only happens in the case of supplying. We compare the tx_log.address with compound_token address because the transfer log will emit from the compound token, and check tx_log.topics[2] with context.tx_log.topics[2] which means the token is being transfer to the same address for whome the supplying is done.